### PR TITLE
Added torch equivalent of InterAtomicL2Distances in torch_layers.py + YAPF changes

### PR DESCRIPTION
**Pull request by atreyamaj**
# Pull Request Template

## Description
Added torch equivalent of InterAtomicL2Distances in torch_layers.py.

How should I reflect the change in the docs?
Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by arunppsg — CHANGES_REQUESTED** [review]

File: deepchem/models/torch_models/layers.py [hunk 1]
```diff
@@ -851,3 +852,78 @@ def __repr__(self) -> str:
...
+  """Compute (squared) L2 Distances between atoms given neighbors.
+
+  This class is the pytorch implementation of the original InteratomicL2Distances layer implemented in Keras.
+  Pairwise distance (L2) is calculated between input atoms, given the number of neighbors to consider, along with the number of descriptors for every atom.
+
+  Examples
+  --------
+  >>> import numpy as np
+  >>> import deepchem as dc
+  >>> from deepchem.models.torch_models.layers import InteratomicL2Distances
```
[the remark this answers is not in the document]
Comment (atreyamaj) [reply deepchem/models/torch_models/layers.py:867]: I have removed the import statements :+1:

**Comment by rbharath** [comment]
> Some changes not made by me are being shown due to rebasing, please ignore them.

Rebasing should remove all changes that aren't due to this PR. Can you doublecheck to make sure there isn't an error?

**Comment by atreyamaj** [comment]
> > Some changes not made by me are being shown due to rebasing, please ignore them.
> 
> Rebasing should remove all changes that aren't due to this PR. Can you doublecheck to make sure there isn't an error?

I have rebased again, but the new changes are still there.

**Comment by arunppsg** [comment]
LGTM! Merging this in.

**Pull request merged**